users/src/main/java/engineering/everest/lhotse/users/domain/UserAggregate.java

```diff
@@ -87,6 +95,13 @@ void on(UserDetailsUpdatedByAdminEvent event) {
         displayName = selectDesiredState(event.getDisplayNameChange(), displayName);
     }
 
+    @EventSourcingHandler
+    void on(UserDeletedAndForgottenEvent event) {
+        userEmail = "";
+        displayName = "";
```

Comment:
Technically these are not necessary, right? A deleted aggregate object cannot handle any future command and should be GC'd eventually.

Reply:
True, and wiping snapshots takes care of flushing out the final remnants.